### PR TITLE
[jiterator] sigmoid : complex dtypes

### PR DESCRIPTION
**Pull request by kshitij12345**
As per title

Relies on existing OpInfo tests from `test_ops.py` and `test_unary_ufuncs.py`

### CODE REVIEW

File: aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu [hunk 1]
```diff
@@ -120,12 +120,39 @@ void i1e_kernel_cuda(TensorIteratorBase& iter) {
...
+const char sigmoid_name[] = "sigmoid";
 void sigmoid_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.common_dtype(), "sigmoid_cuda", [&]() {
-    gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-      return static_cast<scalar_t>(1) / (static_cast<scalar_t>(1) + std::exp(-a));
+  auto common_dtype = iter.common_dtype();
+  if (at::isComplexType(common_dtype)) {
+    // only jiterate for complex-dtype
+    #if AT_USE_JITERATOR()
+      static const auto sigmoid_string = jiterator_stringify(
```
Comment (anjali411) [inline aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:129]: should we add this in `aten/src/ATen/native/cuda/Math.cuh` along with the other op strings?

**Review by anjali411 — APPROVED** [review]
LGTM but I'll let @mruberry do a pass since it's my first time reviewing one of the jiterator PRs.

File: aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu (hunk 1 above)
Comment (kshitij12345) [reply aten/src/ATen/native/cuda/UnarySpecialOpsKernel.cu:129]: The reason for that was `sigmoid` is a relatively trivial op that does not depend on other operator strings from `Math.cuh`. Also touching `Math.cuh` leads to building of multiple files so kept it here.

**Comment by kshitij12345** [comment]
@mruberry gentle ping :)

**Pull request closed without merging**